src/client/components/common/InputNumberWithUnits/InputNumberWithUnits.tsx

```diff
@@ -0,0 +1,26 @@
+import { FC } from 'react';
+import styles from './InputTextWithUnits.module.scss';
+
+interface NumberInputWithUnitsProps {
+  prefix?: string;
+  suffix?: string;
+  value?: number;
+  onChange?: (value: number) => void;
+  name?: string;
+}
+
+export const NumberInputWithUnits: FC<NumberInputWithUnitsProps> = ({ prefix, suffix, value, onChange, name }) => {
+  return (
+    <div className={styles.container}>
+      {prefix && <div>{prefix}</div>}
+      <input
+        className={styles.input}
+        type="number"
+        value={value}
+        name={name}
+        onChange={(event) => onChange(Number(event.target.value))}
+      />
+      {suffix && <div>{suffix}</div>}
+    </div>
+  );
+};
```

Comment:
UI Text Field supports adornments, perhaps we could use that rather than creating custom component?

https://mui.com/material-ui/react-text-field/#input-adornments

Reply:
Yeah, I considered it, but I have PTSD from a previous job trying to apply custom designs to a pre-styled library. Maybe it would work in this case but it's always kind of brittle and now you have 2 ways to style things, less flexibility, you have to look into MUI docs to change stuff... this seemed simple enough to create from scratch. If it turns out we need more fields like this in the future and MUI would help significantly, we can reconsider.